targets/TARGET_STM/TARGET_STM32F0/analogin_device.c

```diff
@@ -89,9 +87,7 @@ void analogin_init(analogin_t *obj, PinName pin)
         error("Cannot initialize ADC");
     }
 
-    // ADC calibration is done only once
-    if (!adc_calibrated) {
-        adc_calibrated = true;
+    if (!LL_ADC_REG_ReadConversionData6(obj->handle.Instance)) {
```

Comment:
I don't understand how this macro can determine if the calibration has been done or not ?

Reply:
Check RM:
ADC data register (ADC_DR)
Just after a calibration is complete, DATA[6:0] contains the calibration factor